src/ray/raylet/node_manager.cc

```diff
@@ -388,69 +388,74 @@ void NodeManager::Heartbeat() {
   SchedulingResources &local_resources = cluster_resource_map_[self_node_id_];
   heartbeat_data->set_client_id(self_node_id_.Binary());
 
-  // TODO(atumanov): modify the heartbeat table protocol to use the ResourceSet directly.
-  // TODO(atumanov): implement a ResourceSet const_iterator.
-  // If light heartbeat enabled, we only set filed that represent resources changed.
-  if (light_heartbeat_enabled_) {
-    if (!last_heartbeat_resources_.GetAvailableResources().IsEqual(
-            local_resources.GetAvailableResources())) {
+  if (new_scheduler_enabled_) {
+    new_resource_scheduler_->Heartbeat(heartbeat_data, light_heartbeat_enabled_);
+    cluster_task_manager_->Heartbeat(heartbeat_data, light_heartbeat_enabled_);
+  } else {
+    // TODO(atumanov): modify the heartbeat table protocol to use the ResourceSet directly.
+    // TODO(atumanov): implement a ResourceSet const_iterator.
+    // If light heartbeat enabled, we only set filed that represent resources changed.
+    if (light_heartbeat_enabled_) {
```

Comment:
All this must be just same as previous one right?

Reply:
yeah the diff looks pretty bad but all i did was move this into the `else`